runtime/src/main/kotlin/kotlin/native/concurrent/WorkerBoundReference.kt

```diff
@@ -55,12 +57,17 @@ public class WorkerBoundReference<out T : Any>(value: T) {
      */
     val worker: Worker = Worker.current
 
+    private lateinit var cleaner: Cleaner
+
     @ExportForCppRuntime("Kotlin_WorkerBoundReference_freezeHook")
     private fun freezeHook() {
         // If this hook was already run, do nothing.
         if (valueBeforeFreezing == null)
             return
         ptr = createWorkerBoundReference(valueBeforeFreezing!!)
+        cleaner = createCleaner(ptr) {
```

Comment:
It might make some sense to reuse the same lambda object for all WBRs as an optimization.

Reply:
Lambda isn't even needed here. Replaced with function reference.